script/tasks/diamondSyncWhitelist.sh

```
@@ -0,0 +1,755 @@
+      local ATTEMPTS=1
+      while [ $ATTEMPTS -le "$MAX_ATTEMPTS_PER_SCRIPT_EXECUTION" ]; do
+        # Use batchSetContractSelectorWhitelist for efficiency
+        # This function is idempotent - calling it multiple times with same pairs is safe
+        # but inefficient. We call it once with ALL pairs.
+        printf '\033[0;36m%s\033[0m\n' "📤 [$NETWORK] Attempt $ATTEMPTS: Calling batchSetContractSelectorWhitelist with ${#CONTRACT_ADDRESSES[@]} pairs"
+        
+        # Capture transaction output to check if it succeeded
+        local TX_OUTPUT
+        echo "CONTRACTS_ARRAY: $CONTRACTS_ARRAY" >&2
+        echo "SELECTORS_ARRAY: $SELECTORS_ARRAY" >&2
+        TX_OUTPUT=$(cast send "$DIAMOND_ADDRESS" "batchSetContractSelectorWhitelist(address[],bytes4[],bool)" "[$CONTRACTS_ARRAY]" "[$SELECTORS_ARRAY]" true --rpc-url "$RPC_URL" --private-key $(getPrivateKey "$NETWORK" "$ENVIRONMENT") --legacy 2>&1)
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)